shade_janitor/janitor.py

```diff
@@ -30,17 +58,14 @@
     # Initialize and turn on debug logging
     shade.simple_logging(debug=True)
 
-    # Initialize cloud
-    # Cloud configs are read with os-client-config
-    cloud = shade.openstack_cloud(cloud=args.cloud)
+    cloud = initialize_cloud(args.cloud)
 
-    resources = SelectRelatedResources(cloud)
+    resources = Resources(cloud)
     cleanup = {}
 
     if args.old_instances:
-        age_resources = SelectAgeRelatedResources(cloud)
```

Comment:
I think this logic may end up cleaning up all old instances in one pass, which was not the intent here.  used 2 different Selectors to identify old instances and then cleanup the oldest one first.  I don't see necessary changes to drop everything from resources to make this work and maintainable.